src/oci.c

```diff
@@ -394,31 +425,72 @@ cc_oci_kill (struct cc_oci_config *config,
 		if (! cc_oci_state_file_create (config, state->create_time)) {
 			g_critical ("failed to recreate state file");
 		}
-		return false;
+		goto error;
 	}
 #endif //UNIT_TESTING
 
-	if (signum == SIGKILL) {
-		/* cc-shim is not able to catch the exit code of a process
-		 * finished with SIGKILL signal hence cc-shim MUST receive
-		 * this signal too
-		 */
-		if (kill (state->pid, signum) < 0 && errno != ESRCH) {
-			g_critical ("failed to stop cc-shim %u: %s",
-					(unsigned)state->pid,
-					strerror (errno));
-			return false;
+	int tries = 1;
+	/* termination signals */
+	const int term_signals[] = {
+		SIGTERM, SIGINT, SIGQUIT, SIGKILL, SIGHUP, -1
```

Comment:
One cant really say that the signals listed here besides SIGKILL are termination signals, they may very well be handled by the container process in the VM to not exit or be masked completely. SIGHUP may just cause configuration for a process to just be reloaded. 
One can say that a signal is terminating, if the shim itself has received exit code from the container process and exited.